components/dropbox/dropbox.app.mjs

```diff
@@ -0,0 +1,463 @@
+import dropbox from "dropbox";
+import fetch from "isomorphic-fetch";
+import get from "lodash/get.js";
+import config from "./config.mjs";
+import isString from "lodash/isString.js";
+import isEmpty from "lodash/isEmpty.js";
+
+const Dropbox = dropbox.Dropbox;
+
+export default {
+  type: "app",
+  app: "dropbox",
+  propDefinitions: {
+    pathFolder: {
+      type: "string",
+      label: "Path",
+      description: "The folder path. (Please use a valid path to filter the values)",
+      useQuery: true,
+      async options({
+        query,
+        returnSimpleString,
+      }) {
+        return this.getPathOptions(query, {
+          omitFiles: true,
+          returnSimpleString,
+        });
+      },
+    },
+    pathFile: {
+      type: "string",
+      label: "Path",
+      description: "The file path. (Please use a valid path to filter the values)",
+      useQuery: true,
+      async options({
+        query,
+        returnSimpleString,
+      }) {
+        return this.getPathOptions(query, {
+          omitFolders: true,
+          returnSimpleString,
+        });
+      },
+    },
+    pathFileFolder: {
+      type: "string",
+      label: "Path",
+      description: "The file or folder path. (Please use a valid path to filter the values)",
+      useQuery: true,
+      async options({
+        query,
+        returnSimpleString,
+      }) {
+        return this.getPathOptions(query, {
+          returnSimpleString,
+        });
+      },
+    },
+    fileRevision: {
+      type: "string",
+      label: "Path",
+      description: "The file revision",
+      async options({ path }) {
+        return this.getFileRevisionOptions(get(path, "value", path));
+      },
+    },
+    recursive: {
+      type: "boolean",
+      label: "Recursive",
+      description: "Also watch sub-directories and their contents.",
+      default: false,
+    },
+    query: {
+      type: "string",
+      label: "Query",
+      description: "The string to search for. May match across multiple fields based on the request arguments.",
+    },
+    limit: {
+      type: "integer",
+      label: "Limit",
+      description: "Specify a max amount of register to be fetched.",
+      optional: true,
+      min: 1,
+    },
```

Comment:
I'd include a note about what this will default to:
`description: "Specify a max amount of register to be fetched. Defaults to 100 if left blank.",`